packages/Switch/src/Switch.styles.js

```diff
@@ -0,0 +1,158 @@
+import styled from "styled-components";
+import tokens from "@paprika/tokens";
+import stylers from "@paprika/stylers";
+
+const KnobPositionStyles = {
+  small: `
+    transform: translateX(12px);
+  `,
+  medium: `
+    transform: translate(20px, 0);
+  `,
+  large: `
+    transform: translate(28px);
+  `,
+};
+
+const KnobSizeStyles = {
+  small: `
+    height: 12px;
+    width: 12px;
+
+    &::after {
+      height: 13px;
+    }
+  `,
+  medium: `
+    height: 20px;
+    width: 20px;
+
+    &::after {
+      height: 21px;
+    }
+  `,
+  large: `
+    height: 28px;
+    width: 28px;
+
+    &::after {
+      height: 29px;
+    }
+  `,
+};
+
+const UnderlaySizeStyles = {
+  small: `
+    height: ${stylers.spacer(2)};
+    width: 28px;
+  `,
+  medium: `
+    height:  ${stylers.spacer(3)};
+    width: 44px;
+  `,
+  large: `
+    height:${stylers.spacer(4)};
+    width: 60px;
+  `,
+};
+
+export const KnobStyled = styled.span`
+  background-color: ${tokens.color.white};
+  border-radius: 100%;
+  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
+  display: inline-block;
+  left: 2px;
+  position: absolute;
+  top: 2px;
+  transition: transform 0.25s ease;
+`;
+
+export const UnderlayStyled = styled.span`
+  background-color: ${tokens.color.blackLighten50};
+  border-radius: 50px;
+  display: inline-block;
+  position: absolute;
+  transition: 0.25s ease-in-out;
+  user-select: none;
+  -moz-user-select: none;
+  -khtml-user-select: none;
+  -webkit-user-select: none;
+  -ms-user-select: none;
+  vertical-align: middle;
+`;
+
+const SwitchStyles = ({ isDisabled, size }) => `
```

Comment:
This came up recently in my PR, and we settled on a different approach here for dealing with props, even though this _IS_ a nice clean way of implementing.  Instead we will go with something like:

```js
const switchStyles = css`
  ${({isDisabled}) => isDisabled ? disabledStyles : null}
`;
```

In fact, it helps with syntax highlighting and linting if we just use the `css` helper for all our js styles.

Reference: https://github.com/acl-services/paprika/pull/92#pullrequestreview-252441383